aiobungie/internal/factory.py

```diff
@@ -124,14 +124,14 @@ def deserialize_destiny_membership(
             id=int(payload["membershipId"]),
             name=name,
             code=payload.get("bungieGlobalDisplayNameCode", None),
-            last_seen_name=payload.get("LastSeenDisplayName", payload["displayName"]),
+            last_seen_name=payload.get("LastSeenDisplayName", payload.get("displayName", name)),
```

Comment:
You cant set the default to `name` since its either `Undefined` or `str`. `last_seen_name` must be a `str`.

This is a quick fix for now.
```suggestion
            last_seen_name=payload.get("LastSeenDisplayName") or payload.get("displayName") or ""
```